CMakeLists.txt

```diff
@@ -105,6 +105,8 @@ set(UTILS_SOURCES
     "${UTILS_DIR}/byte_ring.h"
     "${UTILS_DIR}/serialization.c"
     "${UTILS_DIR}/serialization.h"
+    "${UTILS_DIR}/timestamp.cpp"
+    "${UTILS_DIR}/timestamp.hpp"
```

Comment:
File names should match the class name `TimestampState`. Not 100% sold on `TimestampState` but not sure what else it should be. It really is just a timestamp but that's already taken.

Reply:
I was thinking the same thing, TimestampState was kind of weird. But I didn't know what else to call it because yeah, timestamp is already taken. Though TimestampState does sort of make sense because that's what it is: a class storing the state of a timestamp. 

Maybe TimestampManager? Or something along those lines? I will change the filenames though.

Reply:
What do you think of the new name (TimestampManager)?